### DISCUSSION TITLE
Suggestion to keep the getters in the core contracts

### DISCUSSION THREAD

**Discussion by PaulRBerg** [sablier-labs/evm-monorepo#59] [Ideas]
It was suggested in https://github.com/sablierhq/v2-core/discussions/34 to move the constant getters to the periphery, and we sort of reached a rough consensus on that.

But after pondering this for a while, I think that it would be better to keep the constant getters in the core contracts (`SablierV2Linear` etc.). Here's my rationale:

1. The bespoke getters for each struct member will be very useful for third-party integrations and devs. Asking them to store an additional periphery contract's address just for accessing some basic data about the core contract's streams adds overhead to the complexity of integrating Sablier.
2. Making a query through a periphery contract that further calls the core contract would be more expensive gas-wise than just calling the core contract.
3. As per the discussion in https://github.com/sablierhq/v2-core/discussions/52, we might end up using [PRBProxy](https://github.com/paulrberg/prb-proxy) instead of building customized periphery contracts.

**Comment by razgraf** [comment]
I concur, atomic getters are fine being stored in the individual contracts.
But any complex getter that might be used later for aggregating data (random example if we ever need it: `getFullStreamWithActiveSegments()` ) are a bit more opinionated and case-specific so those I'd keep in a helper periphery contract.

**Reply by PaulRBerg** [reply]
Yup, very much agree with this approach.

**Comment by PaulRBerg** [comment]
Locking this discussion since we reached rough consensus on this.
